propagators/opentelemetry-propagator-opentracing/src/OpenTracingPropagator.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Baggage,
+  Context,
+  getBaggage,
+  getSpanContext,
+  isSpanContextValid,
+  isValidSpanId,
+  isValidTraceId,
+  setBaggage,
+  setSpanContext,
+  TextMapGetter,
+  TextMapPropagator,
+  TextMapSetter,
+  TraceFlags,
+} from '@opentelemetry/api';
+
+/** OT header keys */
+export const OT_TRACE_ID_HEADER = 'ot-tracer-traceid';
+export const OT_SPAN_ID_HEADER = 'ot-tracer-spanid';
+export const OT_SAMPLED_HEADER = 'ot-tracer-sampled';
+export const OT_BAGGAGE_PREFIX = 'ot-tracer-baggage-';
+
+const FIELDS = [OT_TRACE_ID_HEADER, OT_SPAN_ID_HEADER, OT_SAMPLED_HEADER];
+const PADDING = '0'.repeat(16);
+
+function readHeader(
+  carrier: unknown,
+  getter: TextMapGetter,
+  key: string
+): string {
+  let header = getter.get(carrier, key);
+  if (Array.isArray(header)) [header] = header;
+  return header || '';
+}
+
+/**
+ * Propagator for the OpenTracing HTTP format.
+ */
+export class OpenTracingPropagator implements TextMapPropagator {
+  inject(context: Context, carrier: unknown, setter: TextMapSetter) {
+    const spanContext = getSpanContext(context);
+    if (!spanContext || !isSpanContextValid(spanContext)) return;
+
+    setter.set(carrier, OT_TRACE_ID_HEADER, spanContext.traceId.substr(16));
+    setter.set(carrier, OT_SPAN_ID_HEADER, spanContext.spanId);
+    setter.set(
+      carrier,
+      OT_SAMPLED_HEADER,
+      spanContext.traceFlags === TraceFlags.SAMPLED ? 'true' : 'false'
+    );
+
+    const baggage = getBaggage(context);
+    if (!baggage) return;
+
+    Object.entries(baggage).forEach(([k, v]) => {
+      setter.set(carrier, `${OT_BAGGAGE_PREFIX}${k}`, v.value);
```

Comment:
Should we do some checking that this results in valid HTTP header names?
What about corner cases with key which differ only in casing? Should we serialize them?

Reply:
I've updated this to drop keys and values that would be invalid for HTTP transport as that could cause an HTTP request to fail. Case-sensitivity will depend on use case. HTTP should treat these as case insensitive, but a carrier can be used for other purposes. For this reason, I'm leaving the casing as-is.

Reply:
usually I would recommend to log dropped entries but propagators have no logger as far as I know.

Maybe add a note in readme that such entries are silently discarded.

Reply:
I added a note.